cdc/sink/mysql.go

```diff
@@ -348,8 +353,26 @@ func needSwitchDB(ddl *model.DDLEvent) bool {
 	return true
 }
 
+func (s *mysqlSink) adjustSQLModeCompatible(ctx context.Context) error {
+	var oldMode, newMode string
+	row := s.db.QueryRowContext(ctx, "SELECT @@SESSION.sql_mode;")
```

Comment:
`s.db` is a `sql.DB` so some session-level changes may get lost after the inner connection is released to the connection pool of `sql.DB` in my memory